### PR TITLE
[ERE-930] Registered Clinician - Create Patient

### PR DESCRIPTION
**Pull request by ScriptSmith**

### CODE REVIEW

File: rdrf/rdrf/views/patient_view.py [hunk 1]
```diff
@@ -336,54 +342,73 @@ def get_form_kwargs(self):
...
+                if doctor_formset:
+                    doctor_formset.instance = self.object
+                    doctor_formset.save()
+
+            # create user
+            if self.registry_model.has_feature(RegistryFeatures.PATIENTS_CREATE_USERS):
+                user = CustomUser.objects.create(
+                    email=self.object.email,
+                    username=self.object.email,
+                    force_password_change=True,
```
Comment (ScriptSmith) [inline rdrf/rdrf/views/patient_view.py:380]: Synchronise working groups here?

File: rdrf/rdrf/urls.py [hunk 2]
```diff
@@ -312,8 +313,9 @@ def proms_only(url_pattern):
             name='registration_activation_complete'),
 
     re_path(r'^activate/(?P<activation_key>\w+)/?$',
-            ActivationView.as_view(),
+            PatientActivationView.as_view(),
             name='registration_activate'),
+    re_path(r'^resend-activation/?$', ResendActivationView.as_view(), name='registration_resend_activation'),
```
Comment (ScriptSmith) [inline rdrf/rdrf/urls.py]: Worth adding? I'm not sure if there are security concerns with allowing patients to initiate re-activation.

Comment (alintreznai) [reply rdrf/rdrf/urls.py]: I think it's fine to have this as a backup but only admins/clinicians should be able to resend the activation mail for a patient.

File: rdrf/rdrf/views/patient_view.py (hunk 1 above)
Comment (alintreznai) [reply rdrf/rdrf/views/patient_view.py:380]: Yes.

File: rdrf/registry/patients/admin_forms.py [hunk 3]
```diff
@@ -514,8 +516,16 @@ def clean(self):
...
         self._validate_custom_consents()
 
+        self._validate_patient_users()
+
         return super().clean()
 
+    def _validate_patient_users(self):
+        for registry in self.cleaned_data["rdrf_registry"]:
+            if registry.has_feature(RegistryFeatures.PATIENTS_CREATE_USERS):
+                if CustomUser.objects.filter(username=self.cleaned_data["email"]).exists():
```
Comment (alintreznai) [inline rdrf/registry/patients/admin_forms.py]: Maybe use `username__iexact` ?

**Pull request merged**